### ISSUE TITLE
Version >= 6.6.0 does not work correctly with MSG100

### ISSUE THREAD

**Issue by emcepe** [donavanbecker/homebridge-meross#336]
Since version 6.6.0 of this plugin, HomeKit will show my garage door as open after closing it manually with an external button even if it's closed. If I press the icon in iOS HomeKit to close it, it disappears immediately. v6.5.0 and before do not have this problem. I have the MSG100HK module.

I am using Homebridge v1.3.4, UI-X is installed, as well as several other plugins, all at newest version.

Running on a Raspberry Pi 4b with Raspian Buster.

Node.js is v14.17.5
Npm is v6.14.14

Thank you.

**Comment by bwp91** [comment]
hi @emcepe 

do you have this device setup as a cloud device or a local device? have you changed any cloud/local polling rates in the configuration?

it would be useful if you could put the plugin in debug mode via the plugin settings, and post here all/any log entries that appear from closing the garage door with the button, and if the device is setup as local, some polling entries from before and after you close the garage door with the button.

Hope that makes sense!

**Comment by emcepe** [comment]
Hi @bwp91,

it's set up as local device.
I did not change any polling rates or similar, just everything default.

Debugging stuff has to wait until next week because I am busy today and tomorrow and on vacation over the weekend.

**Comment by bwp91** [comment]
Okay thanks.
Before I can do anything I need to see the logs to see exactly what’s going on.
Shall wait to hear from you

**Comment by Gandalf8266** [comment]
I've the same Problem, the Garage door opener MSG100 does not work since one of the last updates. I didn't change anything of my configuration.

**Comment by bwp91** [comment]
> I've the same Problem, the Garage door opener MSG100 does not work since one of the last updates. I didn't change anything of my configuration.

What exactly doesn't work? Also what does work? It's hard for me to pinpoint the issue without knowing where to look!

**Comment by Gandalf8266** [comment]
Nothing does work, when I push the Button in the home app, the Button goes active, but the Garagedor won't open.
When I push the button again, the status is inactive, but nothing happens.

**Comment by bwp91** [comment]
@Gandalf8266 could you update to beta version 6.9.1-beta.0 and see if this works?

**Comment by Gandalf8266** [comment]
I figured out, that the configuration has changed, my configuration was with "message id", "timestamp" and "sign".
Now I need my meoss login. 
does I need anything else? I found a section in the configuration with "Garage" there is needed a name and a Serial Nummer UUID, is this mandatory?

**Comment by bwp91** [comment]
> I figured out, that the configuration has changed, my configuration was with "message id", "timestamp" and "sign".
> Now I need my meoss login.
> does I need anything else? I found a section in the configuration with "Garage" there is needed a name and a Serial Nummer UUID, is this mandatory?

Yes, the plugin now requires either meross credentials or a user key (which can be obtained by inserting credentials into the config and checking the logs in debug mode)

The garage devices config section can be used to enter specific things for the device including an IP for local control and a garage door opening time.

It’s easiest to use the homebridge UI to configure these settings

Using the UI will also remove any no longer needed configuration items.

**Comment by Gandalf8266** [comment]
thx, it works, but the icon shows the wrong state, when I open and close the door with the meross app.
I try to explain:
- Status in home app is closed
- I open with meross app
- Status in home app is open
- I close the door with meross app
- Status in home app keeps "open"
- when I push the button in the home app, it switches to closed an the door keeps closed.

**Comment by emcepe** [comment]
That's exactly what I wrote in my initial issue comment :)

**Comment by bwp91** [comment]
@Gandalf8266 do you have the device setup as a cloud device (this is the default) or a local device?

**Comment by Frankdoc** [comment]
I have the exact same issue as @Gandalf8266 states
@bwp91 I have it setup as a cloud device

**Comment by bwp91** [comment]
Please could one of you put the plugin into debug mode, and obtain the logs that appear from a cycle of opening and closing the garage door?

**Comment by Frankdoc** [comment]
@bwp91
 `
[8/27/2021, 1:40:30 PM] [Meross] [Poort] incoming mqtt [{"header":{"triggerSrc":"DevicePysical","timestampMs":947,"timestamp":1630064430,"sign":"4469d5dc6bee360635ee7cfbc3b7628f","payloadVersion":1,"namespace":"Appliance.GarageDoor.State","method":"PUSH","messageId":"6f377442f9553ee197568b62f4dcf5d9","from":"/appliance/1909182089075490801948e1e9520341/publish"},"payload":{"state":[{"open":1,"lmTime":1630064429,"channel":0}]}}]: %s.
[8/27/2021, 1:40:30 PM] [Meross] [Poort] failed to refresh status as Cannot read property 'getInitialTime' of undefined at deviceGarageSingle.applyUpdate (/homebridge/node_modules/homebridge-meross/lib/device/garage-single.js:460:59).
[8/27/2021, 1:41:15 PM] [Meross] [Poort] incoming mqtt [{"header":{"triggerSrc":"DevicePysical","timestampMs":22,"timestamp":1630064475,"sign":"0aac67b15e31a94353725e754f22e0f5","payloadVersion":1,"namespace":"Appliance.GarageDoor.State","method":"PUSH","messageId":"1e413f57d462fb2593d5813d7c40d45f","from":"/appliance/1909182089075490801948e1e9520341/publish"},"payload":{"state":[{"open":0,"lmTime":1630064474,"channel":0}]}}]: %s.

`

This is when using the manual button. When the garage door is closed the status in homekit remains 'open'. Afterwards I clicked on it and it changed to closed after a second.

**Comment by bwp91** [comment]
@Frankdoc 
could you update to the beta version of the plugin and do this again? The error that’s shown in your logs should have been fixed in the beta I pushed earlier.

**Comment by Frankdoc** [comment]
@bwp91 I installed the beta but now nothing is working anymore. When pressing manual buttons nothing shows in the logs

**Comment by bwp91** [comment]
@Frankdoc are there any warnings/errors in the log when the plugin starts?

**Comment by Frankdoc** [comment]
@bwp91 

[8/27/2021, 1:55:12 PM] [Meross] Disabling cloud client as Cannot read property 'httpRetry' of undefined at connectionHTTP.login (/homebridge/node_modules/homebridge-meross/lib/connection/http.js:90:55).
[8/27/2021, 1:55:12 PM] [Meross] ***** Disabling plugin. *****
[8/27/2021, 1:55:12 PM] [Meross] ***** Neither of username and password nor user key has been configured. *****

*edit : resaved settings and restarted HB and now it seems to work fine.

[8/27/2021, 2:25:05 PM] [Meross] [Poort] incoming mqtt [{"header":{"triggerSrc":"DevicePysical","timestampMs":648,"timestamp":1630067105,"sign":"e57db528785d2ed647d243aba6e16ae3","payloadVersion":1,"namespace":"Appliance.GarageDoor.State","method":"PUSH","messageId":"f9f0746791941f27e0e4fe9fc98ec038","from":"/appliance/1909182089075490801948e1e9520341/publish"},"payload":{"state":[{"open":1,"lmTime":1630067104,"channel":0}]}}]: %s.
[8/27/2021, 2:25:05 PM] [Meross] [Poort] target state [open].
[8/27/2021, 2:25:05 PM] [Meross] [Poort] current state [open].
[8/27/2021, 2:25:29 PM] [Meross] [Poort] incoming mqtt [{"header":{"triggerSrc":"DevicePysical","timestampMs":683,"timestamp":1630067129,"sign":"67ad857a1cc8ef7418ad6f2b2148d8f3","payloadVersion":1,"namespace":"Appliance.GarageDoor.State","method":"PUSH","messageId":"d442f5f6e8622e6c7facc82d008a2812","from":"/appliance/1909182089075490801948e1e9520341/publish"},"payload":{"state":[{"open":0,"lmTime":1630067128,"channel":0}]}}]: %s.
[8/27/2021, 2:25:29 PM] [Meross] [Poort] target state [closed].
[8/27/2021, 2:25:29 PM] [Meross] [Poort] current state [closed].

**Comment by bwp91** [comment]
@Frankdoc thanks, can you try 6.9.1-beta.1? keep an eye on the logs when the plugin restarts in case of any other warnings/errors.

**Comment by bwp91** [comment]
Ah just seen your edit!

**Comment by bwp91** [comment]
Just released v6.9.1, which appears to fix this issue :)

**Comment by Gandalf8266** [comment]
Hello, I configered it as a local devcice, with current update, it seems to work correctly
Thx.

**Comment by emcepe** [comment]
Pretty cool. Thanks for the quick fix :)

I'll test it too but not until next week.

Have a nice weekend.

**Comment by emcepe** [comment]
Works for me too.

Thank you.
